controllers/data.controller.js

```diff
@@ -256,14 +256,21 @@ exports.sendQuestionToFrontend = async (req, res) => {
         .status(httpStatus.NOT_FOUND)
         .json({ error: "No questions found for this tier" });
     }
+    let multipleChoices = [];
+    let tries = 0;
+    let selectedQuestion;
+    // tries to avoid infinite loops
+    while (multipleChoices.length < 4 && tries < 5) {
+      selectedQuestion =
+        questions[Math.floor(Math.random() * questions.length)];
 
-    const selectedQuestion =
-      questions[Math.floor(Math.random() * questions.length)];
+      multipleChoices = await getOtherMultipleChoiceAnswers(
+        selectedQuestion,
+        currentTier,
+      );
+      tries += 1;
+    }
```

Comment:
please consider this: If after 5 tries no suitable questions are found, return an error